### PR TITLE
fix: wrong references crashes library

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR improves the maintainability for input processosr by only moving from integration test to unit test as well as fixing a bug that crashes the library if references are wrong

**Related issue(s)**
Related to https://github.com/asyncapi/generator-model-sdk/issues/64

### CODE REVIEW

**Comment by jonaslagoni** [comment]
@derberg I would really like for coveralls to give me an indication here, to ensure that the coverage did not drop with these test changes. 

Can we try switching from trigger `pull_request_target` to `pull_request`? 🤔

**Comment by jonaslagoni** [comment]
Alright then.

**Comment by derberg** [comment]
It works 🎉 
🤝 congrats

Happy with the solution so far? I guess we need a couple of weeks of testing before we roll it out to others? 

`Check test coverage / Build` should be made mandatory.

if we would have `pull_request`, the bot would have no right to make the comment that you can see above

File: src/processors/JsonSchemaInputProcessor.ts [hunk 1]
```diff
@@ -58,19 +58,26 @@ export class JsonSchemaInputProcessor extends AbstractInputProcessor {
...
     if (refParser.$refs.circular && typeof parsedSchema !== 'boolean') {
-      const circularOption : $RefParser.Options = {
-        continueOnError: true,
-        dereference: { circular: true },
-      };
-      await refParser.dereference(localPath, parsedSchema as $RefParser.JSONSchema, circularOption);
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      deRefOption.dereference!.circular = true;
+      Logger.debug(`Input contains circular references. Need to dereference again to associate correct references, using option ${JSON.stringify(deRefOption)}.`);
+      await refParser.dereference(localPath, parsedSchema as $RefParser.JSONSchema, deRefOption);
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/processors/JsonSchemaInputProcessor.ts]: When I initially implemented it, I couldn't get it to work any other way and adapted how we did in the parser. Retried it and it worked out of the box, so I have removed the duplicate parsing

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/processors/JsonSchemaInputProcessor.ts]: > Retried it and it worked out of the box, so I have removed the duplicate parsing

something like `works for me` 🤣

**Review by magicmatatjahu — APPROVED** [review]
LGTM :)

**Comment by jonaslagoni** [comment]
> if we would have `pull_request`, the bot would have no right to make the comment that you can see above

@derberg I played around with switching between `pull_request` and `pull_request_target`. And the comment is not updated (the bot update the comment when commits are pushed) when we are using `pull_request_target`. In order for the bot to write on PR's we have to use `pull_request`. As I have switched to currently.

**Comment by derberg** [comment]
> And the comment is not updated (the bot update the comment when commits are pushed)

I don't get it, you mean the comment from coveralls bot showed up only when you were using `pull_request`?

**Comment by jonaslagoni** [comment]
> > 
> 
> I don't get it, you mean the comment from coveralls bot showed up only when you were using `pull_request`?

Yes, I have switched back and forth figuring out when it updates the comment. I tried to switch here https://github.com/asyncapi/generator-model-sdk/pull/180/commits/f0514edd10ec9abba35d1065ced3f770b1aa6f14
and I assume that only then the bot comment showed up. Also why use `pull_request_target` if the user can switch to `pull_request` and it runs anyway 🤷

**Comment by derberg** [comment]
it runs, it will be, but GITHUB_TOKEN has only read rights as when user changes to `pull_request` then the workflow file is taken from the fork, and token has read rights only to not enable token leaking. `pull_request_target` means that on PR, workflow is taken from master (default branch) and therefore token has more privileges as it cannot be leaked.

This is why it is super confusing for me, that `pull_request` works and bot makes comment 🤷🏼 magic 😄 or the coveralls bot just like sonarcloud bot are completely separate from the workflow, they work independently. So yeah, if `pull_request` works, then just merge 😄 and I put this case into my x-files box of unsolved cases and someday talk to Mulder and Scully to help me out figuring out what was wrong 😆

**Pull request merged**